fastify.js

```diff
@@ -271,15 +271,12 @@ function build (options) {
   }
 
   function _handleTrustProxy (req) {
-    req.ip = proxyAddr(req, proxyFn)
-    req.ips = proxyAddr.all(req, proxyFn)
-    if (req.ip !== undefined) {
-      req.hostname = req.headers['x-forwarded-host']
-    }
+    var ip = proxyAddr(req, proxyFn)
+    return { ip: ip, ips: proxyAddr.all(req, proxyFn), hostname: ip !== undefined ? req.headers['x-forwarded-host'] : undefined }
   }
 
   function _ipAsRemoteAddress (req) {
-    req.ip = req.connection.remoteAddress
+    return { ip: req.connection.remoteAddress, ips: undefined, hostname: undefined }
```

Comment:
Can you refactor this code so it does not allocate any new object? This is a super-hot path.

Reply:
Thank you, @mcollina. Refactored the logic so that it does not create new object.  43657f8